### PR TITLE
Evaluate and improve CAG pattern testing coverage

### LINKED ISSUE #2489: Evaluate and improve CAG pattern testing coverage
### Environment Details

Please indicate the following details about the environment in which you found the bug:

* SDV version:
* Python version:
* Operating System:

### Error Description
We should evaluate our testing coverage of the new CAG patterns to ensure we don't introduce regressions that were already addressed in the original constraints. This issue is scoped to just adding new tests to improve coverage, including:
- Missing unit tests for synthesizer methods added to support CAG integration
- Missing end-to-end synthesizer integration tests for some CAG patterns
- Ensuring that all existing constraint unit/integration tests are ported over to analogous tests for the new CAG pattern 

For any failing tests that do not have simple solutions, a new issue should be raised to address the bug.

### Additional Context
- See [this doc](https://docs.google.com/document/d/10Z5yNcrLwUrrv2jaVb7kuZpxPN3hUA084SK0dwFggUo/edit?tab=t.0#heading=h.syilxuyxgt50) for edge cases specific to `Inequality` that we should make sure are being covered by our tests

### PR DESCRIPTION
**Pull request by fealho**
CU-86b4u4gbg, Resolve #2489.

Issues found:
* Issue #2275 needs to be implemented for Range constraint
* There is no validation that the transform data matches the metadata
* Nans are not reversed correctly by Range/Inequality
* Range/Inequality don’t support datetime columns of different dtypes ('object' and 'pd.datetime')
* Range/Inequality ignore time component of second column if the first one is only a date
* Range/Inequality don't support timezones for 'object' dtype
* Range/Inequality don't support only one column having a timezone

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86b4u4gbg SDV - Evaluate and improve CAG pattern testing coverage #2489](https://app.clickup.com/t/86b4u4gbg)

File: tests/utils.py [hunk 1]
```diff
@@ -143,3 +145,19 @@ def run_pattern(pattern, data, metadata):
...
+    synthesizer = GaussianCopulaSynthesizer(metadata)
+    synthesizer.add_cag(patterns=pattern)
+    synthesizer.fit(data)
+
+    return synthesizer
+
+
+def run_hma(data, metadata, pattern):
+    synthesizer = HMASynthesizer(metadata)
+    synthesizer.add_cag(patterns=[pattern])
```
Comment (frances-h) [inline tests/utils.py:160]: minor: can we have both of these test helpers take the same type of argument for `pattern` for consistency? Also can we rename it to `patterns` or `pattern_list` if it remains a list type?

**Review by frances-h — APPROVED** [review]

File: tests/utils.py [hunk 2]
```diff
@@ -143,3 +145,19 @@ def run_pattern(pattern, data, metadata):
     reverse_transformed = pattern.reverse_transform(transformed)
 
     return updated_metadata, transformed, reverse_transformed
+
+
+def run_copula(data, metadata, patterns):
```
Comment (amontanez24) [inline tests/utils.py:150]: we should rename these to include the fact that they add constraints since they are in the general test utils. Other option is to make it optional to add the constraints

File: tests/utils.py [hunk 3]
```diff
@@ -143,3 +145,19 @@ def run_pattern(pattern, data, metadata):
...
+    synthesizer = GaussianCopulaSynthesizer(metadata)
+    synthesizer.add_cag(patterns=patterns)
```
Comment (amontanez24) [inline tests/utils.py:152]: also can we use the term constraint to replace pattern anywhere that is new code?

**Review by amontanez24 — CHANGES_REQUESTED** [review]

File: tests/utils.py (hunk 3 above)
Comment (fealho) [reply tests/utils.py:152]: @R-Palazzo already does this in his PR: https://github.com/sdv-dev/SDV/pull/2538/files.

**Review by amontanez24 — APPROVED** [review]
